src/MultiDropdownInputWithSearch/MultiDropdownInputWithSearch.tsx

```diff
@@ -0,0 +1,217 @@
+      </div>
+    </Popover>
+  );
+};
+
+function getOptionsWithLabels(options: any, labelKey?: string) {
+  if (labelKey) {
+    return options.map((option: any) => {
+      option.label = option[labelKey];
+      option.value = option[labelKey];
+      return option;
+    });
+  } else {
+    return filter(options, (option: any) => option).map((option: any) => ({
+      label: option,
+      value: option,
+    }));
+  }
+}
+
+const customStyles: StylesConfig<OptionType, true, GroupBase<any>> = {
+  container: (provided) => {
+    return {
+      ...provided,
+      display: "inline-block",
+    };
+  },
+  control: (provided, state) => {
+    let borderStyles = { border: `1px solid ${styles.background.dark}` };
+
+    if (state.isDisabled) {
+      borderStyles = { border: `1px solid ${styles.background.lines}` };
+    } else if (state.isFocused) {
+      borderStyles = { border: `1px solid ${styles.primary["300"]}` };
+    }
+
+    return {
+      ...provided,
+      cursor: "pointer",
+      borderRadius: "6px",
+      color: state.isDisabled ? styles.text["400"] : `${styles.text["800"]}`,
+      caretColor: `${styles.primary["500"]}`,
+      boxShadow: "none",
+      "&:hover": {
+        border: `1px solid ${styles.primary["300"]}`,
+      },
+      ...borderStyles,
+    };
+  },
+  input: (provided, state) => ({
+    ...provided,
+    margin: 0,
+    padding: 0,
+    // @ts-expect-error unreachable type
+    width: state.selectProps.width,
+  }),
+  placeholder: (provided) => ({ ...provided, margin: 0 }),
+  valueContainer: (provided) => ({
+    ...provided,
+    padding: "5px",
+  }),
+  indicatorsContainer: (provided) => ({
+    ...provided,
+    padding: 0,
+    fill: styles.surface.dark,
+    stroke: styles.surface.dark,
+  }),
+  dropdownIndicator: (provided) => ({
+    ...provided,
+    padding: "0px 5px",
+  }),
+  option: (provided, state) => ({
+    ...provided,
+    cursor: "pointer",
+    backgroundColor: state.isSelected
+      ? styles.primary["50"]
+      : state.isFocused
+      ? styles.background.default
+      : "#fff",
+    ":active": {
+      backgroundColor: styles.primary["50"],
+    },
+    padding: "8px 12px",
+    color: styles.text["800"],
+    position: "relative",
+  }),
+  menu: (provided) => ({
+    ...provided,
+    width: "max-content",
+    maxWidth: "350px",
+    marginTop: "12px",
+    zIndex: styles.layer.second,
+  }),
+};
+
+const MultiDropdownInputWithSearch = <T,>({
+  options,
+  label,
+  labelKey,
+  dropdownKey,
+  onChange,
+  value,
+  isDisabled,
+  width = 50,
+  selectLimit,
+}: MultiDropdownInputWithSearchType<T>) => {
+  const [interimValue, setInterimValue] = useState<T[]>(value);
+  const [shouldUseInterimValue, setShouldUseInterimValue] = useState(false);
+
+  const option = useMemo(
+    () => getOptionsWithLabels(options, labelKey),
+    [options, labelKey]
+  );
+
+  const onMenuOpen = () => {
+    setShouldUseInterimValue(true);
+    setInterimValue(value);
```

Comment:
Can just use interimValue to determine if it should use interim value or not so you don't need to keep track another state.

```
shouldUseInterimValue = !interimValue
```

```suggestion
    setInterimValue(value);
```